src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.cs

```diff
@@ -0,0 +1,2720 @@
+        public void Test_UnaryOperatorExpression_Method_BitwiseNot_System_Int64()
+        {
+            string source = @"
+class A
+{
+    System.Int64 Method()
+    {
+        System.Int64 i = default(System.Int64);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Int64)
+  IInvocationExpression ( System.Int64 A.Method()) (OperationKind.InvocationExpression, Type: System.Int64)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Method_BitwiseNot_System_UInt64()
+        {
+            string source = @"
+class A
+{
+    System.UInt64 Method()
+    {
+        System.UInt64 i = default(System.UInt64);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.UInt64)
+  IInvocationExpression ( System.UInt64 A.Method()) (OperationKind.InvocationExpression, Type: System.UInt64)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Method_BitwiseNot_System_Char()
+        {
+            string source = @"
+class A
+{
+    System.Char Method()
+    {
+        System.Char i = default(System.Char);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Int32)
+  IConversionExpression (ConversionKind.CSharp, Implicit) (OperationKind.ConversionExpression, Type: System.Int32)
+    IInvocationExpression ( System.Char A.Method()) (OperationKind.InvocationExpression, Type: System.Char)
+      Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Method_BitwiseNot_System_Decimal()
+        {
+            string source = @"
+class A
+{
+    System.Decimal Method()
+    {
+        System.Decimal i = default(System.Decimal);
+        return /*<bind>*/~Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree =
```

Comment:
>Test_UnaryOperatorExpression_Method_LogicalNot_CustomType [](start = 20, length = 57)

It would be good to have tests for a scenario when CustomType doesn't have the right user-defined operator.